hooks/fleetHooks.ts

```diff
@@ -39,11 +39,34 @@ export const usePurchaseShip = (shipType: ShipType, waypointSymbol: string) => {
   });
 };
 
+export const useMine = (shipSymbol: string) => {
+  const queryClient = useQueryClient();
+  return useMutation({
+    mutationFn: async () => {
+      const { data } = await fleetApi.extractResources({
+        shipSymbol,
+        extractResourcesRequest: {
+          /* Add survey data here */
```

Comment:
Eventually we'll want to add the ability to survey then mine, but I've been having trouble doing that even on their interactive docs.